File: mobile/services/onboardingService.ts
import { apiRequest } from '@/lib/apiClient';
import type { Member, RegistrationSource } from '@/types';

export type LookupResult =
  | { status: 'found'; member: Pick<Member, 'id' | 'is_guest' | 'app_role'> }
  | { status: 'not_found' };

export async function lookupIdentity(
  identity: string,
  meetingId: string
): Promise<LookupResult> {
  return apiRequest<LookupResult>('/onboarding/lookup', {
    method: 'POST',
    body: { identity, meeting_id: meetingId },
    token: '',
  });
}

export interface RegisterGuestPayload {
  name: string;
  email: string;
  phone?: string;
  source: RegistrationSource;
  meeting_id: string;
}

export interface RegisterGuestResult {
  member: Member;
  token: string;
}

export async function registerGuest(
  payload: RegisterGuestPayload
): Promise<RegisterGuestResult> {
  return apiRequest<RegisterGuestResult>('/onboarding/register', {
    method: 'POST',
    body: payload,
    token: '',
  });
}

File: backend/app/services/onboarding_service.py
from datetime import datetime, timedelta, timezone

from jose import jwt

from app.config import settings
from app.models.onboarding import GuestRegisterIn, GuestRegisterOut, LookupIn, LookupResult


async def lookup_identity(body: LookupIn) -> LookupResult:
    # TODO: query members table by email or phone + club_id
    # Returns status='found' with member metadata, or status='not_found'
    raise NotImplementedError


async def register_guest(body: GuestRegisterIn) -> GuestRegisterOut:
    # TODO: insert member row (is_guest=True, app_role=None)
    # Issue a short-lived guest JWT signed with GUEST_JWT_SECRET
    raise NotImplementedError


def _issue_guest_jwt(member_id: str, club_id: str, meeting_id: str) -> str:
    expire = datetime.now(timezone.utc) + timedelta(hours=settings.guest_token_expire_hours)
    payload = {
        "sub": member_id,
        "role": "guest",
        "club_id": club_id,
        "meeting_id": meeting_id,
        "exp": expire,
    }
    return jwt.encode(payload, settings.guest_jwt_secret, algorithm="HS256")
